### ISSUE TITLE
SMB mounts doesn't show up when using Webdav / Netdrive and iOS app

### ISSUE THREAD

**Issue by Whatever2017ldskfj** [nextcloud/server#5973]
### Steps to reproduce
1. Add SMB external storage.
2. Login with external user account.
3. Only local resources are shown in WebDav/NetDrive and iOS app.

### Expected behaviour
SMB shares are shown.

### Actual behaviour
They are not.

Verified via multiple installs of 12.0, from VHD, from scratch, and upgraded from 10.x and 11.x

Affected upgraded instances worked correctly before upgrading to v12.

### Server configuration
**Operating system**: Linux cloud.bccnw.com 4.4.0-89-generic #112-Ubuntu SMP Mon Jul 31 19:38:41 UTC 2017 x86_64

**Web server:** Apache/2.4.18 (Ubuntu) (apache2handler)

**Database:** pgsql PostgreSQL 9.6.3 on x86_64-pc-linux-gnu, compiled by gcc (Ubuntu 5.4.0-6ubuntu1~16.04.4) 5.4.0 20160609, 64-bit

**PHP version:** 7.0.18-0ubuntu0.16.04.1
Modules loaded: Core, date, libxml, openssl, pcre, zlib, filter, hash, Reflection, SPL, session, standard, apache2handler, redis, mysqlnd, PDO, xml, calendar, ctype, curl, dom, mbstring, fileinfo, ftp, gd, gettext, iconv, imap, intl, json, ldap, exif, mcrypt, mysqli, pdo_mysql, pdo_pgsql, pdo_sqlite, pgsql, Phar, posix, readline, shmop, SimpleXML, smbclient, sockets, sqlite3, sysvmsg, sysvsem, sysvshm, tokenizer, wddx, xmlreader, xmlwriter, xsl, zip, libsmbclient, Zend OPcache

**Nextcloud version:** 12.0.1 RC4 - 12.0.1.3
</details>

**Comment by enoch85** [comment]
I can confirm this, I was the one trying to debug it. First tried on MariaDB, also tried with local users, and with LDAP users, none of them worked. Confirmed to be working in 11.0.3 though, but in 12 it was just dead. :(

This is what I was talking about in IRC @icewind1991 and @blizzz

**Comment by meyeaard** [comment]
I have the same issue.

I have presented external storage via SMB and access this via the WebUI, Nextcloud client, and WebDAV client.

When I have the external storage configured to "Save user password in session" so that the users AD credentials are passed back to the Windows file server, I am unable to access the folder via WebDAV and sync with the Nextcloud client fails with 'Internal Server Error', however access via the  WebUI does function properly.
If I change the external storage security to use a static username and password for the SMB share then it is available via WebDAV and the Nextcloud client operates correctly. I am using this as a very bad workaround. It's not secure to store creds in the DB and it doesn't work for multi-user access where each user might have different ACLs to that folder.

This previously worked correctly on Nextcloud 11, before updating to 12.
I am on the current production release (12.0.1) and php 5.6.30

**Comment by Whatever2017ldskfj** [comment]
PLEASE can we get a fix for this? We have client who would gladly pay for this product, but not with this defect. Please. A fix is needed.

**Comment by SteffenDE** [comment]
I just wondered why my external storage folders don't show up on iOS :(

**Comment by jospoortvliet** [comment]
Hi guys,

Please understand this is not a support forum, this is an engineering platform. 'me too' and 'please fix this soon' comments are not helpful at all.

You can help fix this yourself, or one of our engineers works on it if they have time. And they have time if it is important for our customers.

If you need this for a customer of yours or for your business - I suggest you get in contact with sales so we can work something out.

Running software in your company without a support contract is at your own risk.

**Comment by ChristophWurst** [comment]
I'm closing this ticket because it hasn't received any updates in a long time. If this is still relevant, feel free to re-open or file a new ticket.

Thank you.

**Comment by Whatever2017ldskfj** [comment]
Issue, two years later, remained unaddressed. Why close the ticket if the problem isn't fixed? In addition, it has been repeatedly brought up in the support forums and this ticket linked.

**Comment by ChristophWurst** [comment]
> Why close the ticket if the problem isn't fixed?

-> *I'm closing this ticket because it hasn't received any updates in a long time.*

> In addition, it has been repeatedly brought up in the support forums and this ticket linked.

Would be great if that was actually linked.

**Comment by skjnldsv** [comment]
@Whatever2017ldskfj please link the forum ticket?

**Comment by Whatever2017ldskfj** [comment]
I opened this ticket three years ago and was told we'd need to purchase thousands of dollars in support to get the issue fixed. We've long since given up on using NextCloud entirely, as our sole use involved remote access to SMB shares. It's not the job of users to pay to fix serious defects in usability, or to provide links to years-old forum posts. @enoch85 is very knowledgeable about NextCloud, provides a very valuable fast-deploy VM, and confirmed this issue THREE YEARS ago.

**Comment by MorrisJobke** [comment]
> It's not the job of users to pay to fix serious defects in usability, or to provide links to years-old forum posts. @enoch85 is very knowledgeable about NextCloud, provides a very valuable fast-deploy VM, and confirmed this issue THREE YEARS ago.

Just to make it sure: nobody forces you to use Nextcloud. We provide this software to all for free. If there is a defect and you want to have it fixed, then somebody should look into it. But also this somebody has a life and doesn't need to provide fixes for people in the internet for free. Of course this could happen, because it is seen as valuable for this person as well. 

If you don't want to pay for it, you can also politely ask if somebody is willed to look into it - but again there is no guarantee that it will be delivered except you have a contract with this person.

**Comment by Whatever2017ldskfj** [comment]
In this case we had it deployed and in-use at many locations. And an "update" broke critical functionality. So while no one 'forced' use of NextCloud we were dependent on its SMB feature. This isn't an esoteric, minor part of it. It's a core, major feature. Broken in an update. Confirmed by many others. And our years-long requests for it to be fixed were met with what amounted to "pay up or shut up". So we stopped using it. If that "pay up" had been $200 or $500 we'd have paid. But it was on order of US$2000-US$3000.

**Comment by MorrisJobke** [comment]
> In this case we had it deployed and in-use at many locations. And an "update" broke critical functionality. So while no one 'forced' use of NextCloud we were dependent on its SMB feature. This isn't an esoteric, minor part of it. It's a core, major feature. Broken in an update. Confirmed by many others. And our years-long requests for it to be fixed were met with what amounted to "pay up or shut up". So we stopped using it. If that "pay up" had been $200 or $500 we'd have paid. But it was on order of US$2000-US$3000.

And again: this is still a project were people work **voluntarily** on the code and fix things or provide new features.

**Comment by Whatever2017ldskfj** [comment]
I understand. And, including a major feature like SMB access on which users depend, then issuing several major new versions and countless interim updates in which it is completely broken and demanding thousands of dollars to fix feels like being held hostage. End result was that we moved on from the product. I'm only here today because I got notified that there were new posts on the ticket I opened over three YEARS ago.
